libs/community/langchain_community/llms/cleanlab.py

```diff
@@ -0,0 +1,162 @@
+    """Holds configurations for trustworthy language model. 
+       Available options (model, max_tokens, etc.) with their definitions listed here: 
+       https://help.cleanlab.ai/reference/python/trustworthy_language_model/#class-tlmoptions
+    """
+
+    class Config:
+        extra = "forbid"
+
+    @model_validator(mode="before")
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        cleanlab_api_key = convert_to_secret_str(
+            get_from_dict_or_env(values, "cleanlab_api_key", "CLEANLAB_API_KEY")
+        )
+        values["cleanlab_api_key"] = cleanlab_api_key
+
+        try:
+            from cleanlab_studio import Studio
+
+            studio = Studio(api_key=cleanlab_api_key.get_secret_value())
+            # Check for user overrides in options dict
+            use_options = values["options"] is not None
+            # Initialize TLM
+            cls._client = studio.TLM(
+                quality_preset=values["quality_preset"],
+                options=values["options"] if use_options else None,
+            )
+        except ImportError:
+            raise ImportError(
+                "Could not import cleanlab-studio python package. "
+                "Please install it with `pip install -U cleanlab-studio`."
+            )
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            **{"quality_preset": self.quality_preset},
+            **{"options": self.options},
+        }
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        """Get the default parameters for calling Cleanlab API."""
+        default_params = {
+            "quality_preset": "medium",
+            "max_tokens": 512,
+            "model": "gpt-4o-mini",
+        }
+        return {**default_params}
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "trustworthy_language_model"
+
+    def _generate(
+        self,
+        prompts: List[str],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> LLMResult:
+        """Call Cleanlab endpoint and return response with additional info."""
+
+        responses: List[Dict[str, str]] = self._client.prompt(prompts)
+
+        generations = []
+        for resp in responses:
+            text = resp["response"]
+            trustworthiness_score = resp["trustworthiness_score"]
+            if stop is not None:
+                text = enforce_stop_tokens(text, stop)
+            generations.append(
+                [
+                    Generation(
+                        text=text,
+                        generation_info={
+                            "trustworthiness_score": trustworthiness_score
```

Comment:
heads up this won't be surfaced to users in an LLM because of the str -> str interface

Reply:
since we return `LLMResult` object, i was able to extract/expose the `trustworthiness_score`.

<img width="971" alt="Screenshot 2024-11-01 at 15 12 32" src="https://github.com/user-attachments/assets/edc715a9-fcbc-45b0-9ead-0c65362c82b5">

i saw references to `AIMessage` object used by partner pkgs to expose `response_metadata` (`trustworthiness_score` is also a response metadata).
should i use this return object in our TLM's community integration?